Python/bin/sexVals.py

```diff
@@ -244,93 +247,52 @@ def sexDateToISO(sexDate, intsec=False):
-                                 ' This should not happen. ', sexDate)
-                 
-        # Get the string            
-        ss = "{:02d}".format(ss_int)
-       
-     elif (prec == 1):
-
-        # Round to two decimal places
-        ss = "{:05.2f}".format(fracdd)
-        if ss == "60.00":
-           #print("B Fixing: " + ss, mm_int, hh_int)
-           ss = "00.00"
-           mm_int += 1
-           if mm_int == 60:
-              mm_int = 0
-              hh_int += 1
-              if hh_int == 24:
-                 errorSexVal('With two decimal places, fractional portion of date rounds to "24:00:00.00". ' + \
-                                 ' This should not happen. ', sexDate)
-                   
-     elif (prec == 10):
-
-        # Round to one decimal place
-        ss = "{:04.1f}".format(fracdd)
-        if ss == "60.0": 
-           #print("C Fixing: " + ss, mm_int, hh_int)
-           ss = "00.0"
-           mm_int += 1
-           if mm_int == 60:
-              mm_int = 0
-              hh_int += 1
-              if hh_int == 24:
-                 errorSexVal('With one decimal place, fractional portion of date rounds to "24:00:00.0". ' + \
-                                 ' This should not happen. ', sexDate)
-
-     # Write the string
-     mm = "{:02d}".format(mm_int)
-     hh = "{:02d}".format(hh_int)
-     isodate = m.group(1) + '-' + m.group(3)  + '-' + m.group(4)[0:2] + 'T' + hh + ':' + mm + ':' + ss + 'Z'
-
-     if (ss[0:2] == "60") or (ss[0] == "-"):  # should raise error
-        print ("Bad Date: " + ss + " for ss in date:",  sexDate)
-
-     return (isodate, prec, oldfracdd)
+      d = m.groupdict()
+      year = int(d['year'])
+      month = int(d['month'])
+      day = int(d['day'])
+      oldfracdd = "." + d['fractional_day']
+      seconds = float(oldfracdd) * 86400
+      # precision in millions of a day
+      prec = 10.0**(6 - len(d['fractional_day'])) # >1 for legal mpc
+      if prec >= 1:
+         prec = int(prec)
+
+      # Count the number of different precision values
+      if prec == 1: _countDate1 += 1       # This means 1e-6 day precision
+      if prec == 10: _countDate10 += 1
+      if prec == 100: _countDate100 += 1
+      if prec == 1000: _countDate1000 += 1
+      if prec == 10000: _countDate10000 += 1
+      if prec == 100000: _countDate100000 += 1
+      if prec == 1000000: _countDate1000000 += 1 # This means integer day precission
+      if prec >  1000000: _countDateBig += 1 # Should not be possible
+      if prec < 1: _countDateSmall += 1 # Should not be possible
+
+      if mpc_rounding:
+         # convert to millisecond precision if precision is 1e-6 day or less (i.e. all MPC records)
+         digits = 3
+      elif intsec:
+         # do not report microseconds
+         digits = 0
+      else:
+         # pass through at microsecond precision
+         digits = 6
```

Comment:
Can we do smart rounding when not doing mpc_rounding or intsec? And it may be occasionally beneficial to be able to force microsec rounding, and so it could make sense to allow this through a `microsec` option, similar to what is done for `intsec`. Something like 
```
     elif microsec:
         # report microseconds
         digits = 6
     else:
         # smart rounding
         digits = max(0, 2-log10(prec))
```
give or take whatever the python syntax would be. This would lead to the following:
```
prec(10^-6 d)  prec(sec)  digits
---------------------------------------
0.0001          9e-6       6
0.01            9e-4       4
1               0.09       2
10              0.9        1
100             9          0
1000            90         0 (lower bound)
---------------------------------------
```

Reply:
Yes, that's a good idea. I've implemented the change as you've suggested.